### PR TITLE
MEN-7481: The big cleanup

### PR DESCRIPTION
**Pull request by vpodzime**

### CODE REVIEW

**Comment by vpodzime** [comment]
I guess only the first commit is really worth a proper review.

File: core/src/mender-client.c [hunk 1]
```diff
@@ -403,63 +394,15 @@ mender_client_init(mender_client_config_t *config, mender_client_callbacks_t *ca
...
-    /* Add add-on to the list */
-    if (NULL == (tmp = (mender_addon_instance_t **)realloc(mender_client_addons_list, (mender_client_addons_count + 1) * sizeof(mender_addon_instance_t *)))) {
-        mender_log_error("Unable to allocate memory");
-        if (NULL != addon->exit) {
-            addon->exit();
-        }
-        ret = MENDER_FAIL;
+#ifdef CONFIG_MENDER_CLIENT_INVENTORY
+    if (MENDER_OK != (ret = mender_inventory_init())) {
+        mender_log_error("Failed to inialize the inventory functionality");
```
Comment (larsewi) [inline core/src/mender-client.c:399]:
```suggestion
        mender_log_error("Failed to initialize the inventory functionality");
```

**Review by larsewi — APPROVED** [review]

File: platform/storage/posix/src/mender-storage.c [hunk 2]
```diff
@@ -260,48 +260,6 @@ mender_storage_get_update_state(mender_update_state_t *state, char **artifact_ty
...
-        mender_log_error("Unable to delete device configuration");
-        return MENDER_FAIL;
-    }
-
-    return MENDER_OK;
-}
-
-#endif /* CONFIG_MENDER_CLIENT_CONFIGURE_STORAGE */
-#endif /* CONFIG_MENDER_CLIENT_ADD_ON_CONFIGURE */
-
```
Comment (lluiscampos) [inline platform/storage/posix/src/mender-storage.c:304]: Same here, remove `MENDER_STORAGE_NVS_DEVICE_CONFIG`

File: platform/storage/zephyr/nvs/src/mender-storage.c [hunk 3]
```diff
@@ -225,122 +225,8 @@ mender_storage_delete_deployment_data(void) {
...
-
-    /* Delete device configuration */
-    if (0 != nvs_delete(&mender_storage_nvs_handle, MENDER_STORAGE_NVS_DEVICE_CONFIG)) {
-        mender_log_error("Unable to delete device configuration");
-        return MENDER_FAIL;
-    }
-
-    return MENDER_OK;
-}
-
```
Comment (lluiscampos) [inline platform/storage/zephyr/nvs/src/mender-storage.c:343]: Same, remove `MENDER_STORAGE_NVS_DEVICE_CONFIG`

**Review by lluiscampos — APPROVED** [review]
Great work :broom: 

See few comments below

File: include/mender-inventory.h [hunk 4]
```diff
@@ -24,39 +24,24 @@
...
 /**
- * @brief Mender inventory instance
- */
-extern const mender_addon_instance_t mender_inventory_addon_instance;
-
-/**
- * @brief Mender inventory configuration
- */
-typedef struct {
-    int32_t refresh_interval; /**< Inventory refresh interval, default is 28800 seconds, -1 permits to disable periodic execution */
```
[the remark this answers is not in the document]
Comment (vpodzime) [reply include/mender-inventory.h:41]: Added `uint32_t inventory_update_interval` into `mender_client_config_t`.

**Review by lluiscampos — APPROVED** [review]
Thank you

**Pull request merged**